// react-share-social/dist/App.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const jsx_runtime_1 = require("react/jsx-runtime");
// import logo from './logo.svg';
require("./App.css");
const index_1 = require("./components/index");
const styles_1 = require("@mui/styles");
const useStyles = (0, styles_1.makeStyles)({
    App: {
        backgroundColor: '#282c34',
        color: 'white',
        minHeight: '100vh',
        flexDirection: 'column',
        padding: '25px'
    },
    AppHeader: {
        fontStyle: 'normal',
        fontWeight: 800,
        fontSize: '2.25rem',
        lineHeight: 1.2,
        maxWidth: '933px'
    },
    AppSubHeader: {
        fontSize: '1.125rem',
        lineHeight: 1.2,
        paddingBottom: '16px',
        fontWeight: 400
    },
    Box: {
        paddingBottom: '40px'
    }
});
const style = {
    root: {
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
        borderRadius: 3,
        border: 0,
        boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
        color: 'white',
    },
    copyContainer: {
        border: '1px solid blue',
        background: 'rgb(0,0,0,0.7)',
        display: 'none'
    },
    title: {
        color: 'aquamarine',
        fontStyle: 'italic'
    }
};
function App() {
    const classes = useStyles();
    return ((0, jsx_runtime_1.jsxs)("div", Object.assign({ className: classes.App }, { children: [(0, jsx_runtime_1.jsxs)("div", Object.assign({ className: classes.Box, "data-testid": "multiple-icon-box" }, { children: [(0, jsx_runtime_1.jsx)("header", Object.assign({ className: classes.AppHeader }, { children: "Multiple social icons" })), (0, jsx_runtime_1.jsx)("h1", Object.assign({ className: classes.AppSubHeader }, { children: "Each Social Icon triggers a callback" })), (0, jsx_runtime_1.jsx)(index_1.ShareSocial, { socialTypes: ['facebook', 'twitter', 'line', 'linkedin', 'whatsapp', 'viber', 'telegram', 'reddit', 'instapaper', 'livejournal', 'mailru', 'ok', 'hatena',
                            'email', 'workspace'], url: "http://sifatul.github.io/", onSocialButtonClicked: (buttonName) => {
                            console.log(`${buttonName} clicked`);
                        }, title: "Share and get amazing products" })] })), (0, jsx_runtime_1.jsxs)("div", Object.assign({ className: classes.Box, "data-testid": "customizable-box" }, { children: [(0, jsx_runtime_1.jsx)("header", Object.assign({ className: classes.AppHeader }, { children: "Everything is customizable" })), (0, jsx_runtime_1.jsx)("h1", Object.assign({ className: classes.AppSubHeader }, { children: "Custom styles applicable to the all parts of the UI" })), (0, jsx_runtime_1.jsx)(index_1.ShareSocial, { socialTypes: ['facebook', 'twitter', 'line', 'linkedin', 'whatsapp', 'viber', 'telegram'], url: "http://sifatul.github.io/", title: "Share and get amazing products", onSocialButtonClicked: (buttonName) => {
                            alert(`${buttonName} clicked`);
                        }, style: style })] }))] })));
}
exports.default = App;
